packages/block-editor/src/components/block-patterns/index.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo, useCallback } from '@wordpress/element';
+import { parse } from '@wordpress/blocks';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { ENTER, SPACE } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import BlockPreview from '../block-preview';
+
+function BlockPattern( { pattern, onClick } ) {
+	const { title, content } = pattern;
+	const blocks = useMemo( () => parse( content ), [ content ] );
+
+	return (
+		<div
+			className="block-editor-patterns__item"
+			role="button"
+			onClick={ () => onClick( blocks ) }
+			onKeyDown={ ( event ) => {
+				if ( ENTER === event.keyCode || SPACE === event.keyCode ) {
+					onClick( blocks );
+				}
+			} }
+			tabIndex={ 0 }
+		>
+			<div className="block-editor-patterns__item-preview">
+				<BlockPreview blocks={ blocks } />
+			</div>
+			<div className="block-editor-patterns__item-title">{ title }</div>
+		</div>
+	);
+}
+
+function BlockPatterns( { patterns } ) {
+	const getBlockInsertionPoint = useSelect( ( select ) => {
```

Comment:
Hooks are better at so many levels. I recall how much work was required to pass down selector to action to call it when a callback is executed. Here, you get everything out of the box :)